### ISSUE TITLE
sql: delete user  when database depend on it

### ISSUE THREAD

**Issue by EamonZhang** [cockroachdb/cockroach#17566]
Hi 

**Scenario** 

 Create user and grant role to it and then drop this user

 **Postgresql**  

#Create User
`> create user username ;`

#Grant role to user
`> grant all on database mydb to username;`

#Delete user
```
> drop user username;
ERROR:  role "username" cannot be dropped because some objects depend on it
DETAIL:  privileges for database mydb 

```

**CockroachDB**

#Create User
```
$ cockroach user set username --certs-dir=certs --host=10.2.0.14  --password
Enter password: 
Confirm password:
INSERT 1
$ cockroach user ls --certs-dir=certs --host=10.2.0.14
  ```

#Grant role to user
```
> GRANT CREATE ON database  bank TO username;
GRANT
```

```
> show grants on database bank for username;
+----------+----------+------------+
| Database |   User   | Privileges |
+----------+----------+------------+
| bank     | username | CREATE     |
+----------+----------+------------+
(1 row)
```

#Delete user 
```
$ cockroach user rm username --certs-dir=certs --host=10.2.0.14
> show grants on database bank for username;
+----------+----------+------------+
| Database |   User   | Privileges |
+----------+----------+------------+
| bank     | username | CREATE     |
+----------+----------+------------+
(1 row)
```

nani

...

**Comment by dianasaur323** [comment]
@jordanlewis are we going to be able to make this in by 1.1? If not, can you shift the milestone to 1.2?

**Comment by bdarnell** [comment]
This was fixed in #19095
